server/api/absence-record/absence-record.controller.js

```diff
@@ -73,38 +45,66 @@ exports.validateCreate = function(req, res, next) {
     });
 };
 
+function createStudents(newStudents) {
+  return new Promise(function(resolve, reject) {
+    if (!newStudents.length) return resolve([]);
+    Student.collection.insert(newStudents, {ordered: true}, function(err, ins) {
+      if (err) return reject(err);
+      return resolve(ins.ops);
+    });
+  });
+}
+
 /**
  * Creates a new absence record in the DB.
  * restriction: 'teacher'
  */
 exports.create = function(req, res) {
-  var school = req.body.schoolId;
+  var result = {};
   var existingEntries = _.map(req.body.updates || [], 'entry');
-  if (!req.body.creates) {
-    return newAbsenceRecord({
-      schoolYear: req.body.schoolYear,
-      school: school,
-      date: req.body.date,
-      entries: existingEntries
-    }, res, []);
-  }
   var newStudents = _.map(req.body.creates, 'student');
   var newEntries = _.map(req.body.creates, 'entry');
+  var outreaches = [];
+  // Assign students to be created to validated school.
   _.forEach(newStudents, function(student) {
-    student.currentSchool = school;
+    student.currentSchool = req.school.id;
   });
-  Student.collection.insert(newStudents, {ordered: true}, function(err, ins) {
-    var createdStudents = ins.ops;
-    if (err) return handleError(res, err);
+  var promise = createStudents(newStudents);
+  promise.then(function(createdStudents) {
+    // Fill in missing student for new entries.
     _.forEach(createdStudents, function(student, index) {
       newEntries[index].student = student._id;
     });
-    return newAbsenceRecord({
+    var combinedEntries = [].concat.apply(newEntries, existingEntries);
+    // Outreaches for entries are update with student and added to outreaches.
+    _.forEach(combinedEntries, function(entry) {
+      _.forEach(entry.outreaches, function(outreach) {
+        outreach.student = entry.student;
+        outreaches.push(outreach);
+      });
```

Comment:
This is fine. But I've always been curious if there's a more elegant, and efficient way to go about handling cases in which you do need to use nested forEach statements:
http://stackoverflow.com/questions/21532912/nested-lodash-foreach